app.rb

```diff
@@ -0,0 +1,131 @@
+      end
+    end
+    @parent.show_menu
+  end
+
+  def list_of_books
+    if @books.empty?
+      puts 'No Book In The List.'
+    else
+      puts 'List of Books available.'
+      @books.each { |book| puts "Title: #{book.title} By #{book.author}" }
+    end
+    @parent.show_menu
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    @people.push(Student.new(name: name, age: age, parent_permission: parent_permission))
+    puts
+    puts 'Student is successfully created'
+  end
+
+  def create_person
+    puts 'Do you want to create a Student(1) or Teacher(2)? [ENTER NUMBER]:'
+    person_input = gets.chomp
+    case person_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please add a valid input!'
+    end
+    @parent.show_menu
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    @people.push(Teacher.new(specialization: specialization, name: name, age: age,
+                             parent_permission: parent_permission))
+    puts
+    puts 'Teacher is successfully created'
+    @parent.show_menu
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully'
+    @parent.show_menu
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
+
+    puts 'Enter the number of the book:'
+    book_num = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    puts 'Enter the number of the person:'
+    person_num = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals.push(Rental.new(date, @books[book_num], @people[person_num]))
+    puts 'Rental created successfully'
+    @parent.show_menu
+  end
+
+  def list_of_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    puts 'Rentals:'
+    rentals_found = @rentals.select { |rental| rental.person.id == id }
+    if rentals_found.empty?
+      puts 'No rentals found for the given person ID.'
+    else
+      rentals_found.each do |rental|
+        puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
+      end
+    end
+
+    @parent.show_menu
+  end
```

Comment:
- [ Required ] I think it would be nice if you make sure that the functionality for adding a `Rental` and listing `Rentals` works as expected. Since this functionality is dependent on the `create a person` functionality which was not working before I'll have to mark it as a required change. Please ensure that both `create` and `list` rentals functionality work as expected. I would like to suggest that you refactor your code as shown below for this to work.

## Error when creating a rental
![Screenshot from 2023-10-21 18-30-06](https://github.com/stino-x/PST-Classes/assets/16392046/bb4698dc-fee3-4307-bb44-420573095815)

> I think for these two functionalities to work it would be advisable to refactor the following classes as shown below.

## book.rb

> In the Book class you need to make sure that the `add_rentals` method receives a `date`, and the  `person` renting the book, then create the rental through the `Rental` class constructor as shown below. 

```ruby
require_relative 'rental'

class Book
  attr_accessor :title, :author, :rentals

  def initialize(title, author)
    @title = title
    @author = author
    @rentals = []
  end

  def add_rentals(date, book, person)
    rental = Rental.new(date, self, person)
    @rentals << rental
  end
end

```

## person.rb
> In the Person class you need to make sure that the `add_rentals` method receives a `date`, and the  `book` being rented out, then create the rental through the `Rental` class constructor as shown below. 

```ruby
require_relative 'nameable' # Include the Nameable module
require_relative 'rental'

class Person < Nameable
  attr_accessor :name, :age, :rentals
  attr_reader :id

  def initialize(name: 'Unknown', age: 0, parent_permission: true)
    super()
    @name = name
    @id = rand(1..1000)
    @age = age
    @parent_permission = parent_permission
    @rentals = []
  end

  private

  def of_age?
    @age >= 18
  end

  def add_rentals(date, book, person)
    rental = Rental.new(date, book, self)
    @rentals << rental
  end

  public

  def can_use_services?
    of_age? || @parent_permission
  end

  def correct_name
    @name
  end
end

```

Reply:
@Felix45 while i do appreciate i do think its kind of unfair that youd ask me that after three reviwers didnt , the person class and functionality works very well now and i dont think its grounds to overrequest bearung in mind the time constraint

Reply:
> @Felix45 while i do appreciate i do think its kind of unfair that youd ask me that after three reviwers didnt , the person class and functionality works very well now and i dont think its grounds to overrequest bearung in mind the time constraint

Hi @stino-x, 

I hope you are well, the previous reviewers could not check for the `create_rentals` and `list_rentals` functionality because the `App` was broken. The `create_person` functionality was not working as expected, it's important to note that to create a `rental` you need to select a `person` from a list and this functionality was not working before. Currently, when I run your application and try to create a `Rental` I get the error below. 

## Error when creating a rental
![Screenshot from 2023-10-21 18-30-06](https://github.com/stino-x/PST-Classes/assets/16392046/908078a6-e163-4cd2-b791-05c6a462af37)

I would have loved to approve your project but only if it's working correctly, that's why I gave code suggestions to guide you in implementing this.